launch-checklist.md

```diff
@@ -1 +1,21 @@
+# Replicache Launch Checklist
 
+Before you launch with Replicache in your product, it's a good idea to double-check that you have correctly covered all the small details of integration. This list can help you determine if you might have missed a detail, or deferred and then forgotten about one.
+
+- **JS SDK**
+  - if you wish to change the type of a mutator (eg, the number or type of its arguments) you must choose a new name; Replicache does not handle mutator versioning
+  - if you wish to version the schema of the mutations in `Push` and the `clientView` returned in `Pull`, version information is included in `ReplicacheOptions.schemaVersion`
+  - be sure that re-auth is handled for `Push` and `Pull` via `getPushAuth` and `getPullAuth`
+- **All endpoints**
+  - ensure the user is authenticated and the `clientID` verified to belong to that user
+- **Push endpoint**
+  - ensure that the `lastMutationID` for a client is updated transactionally along with the pushed mutations' effects
+  - all mutations with `id`s less than the client's current `lastMutationID` must be skipped
+  - all mutations with `id`s greater than the client's current `lastMutationID+1` must be ignored
+  - if a temporary error is encountered that might be resolved on retry, the endpoint must halt processing mutations and return
+  - if a permanent error is encountered such that the mutation will never be appiable, the endpoint must ignore that mutation and increment the `lastMutationID` as if it were applied. Failing to do this will deadlock the client if a mutation will always fail.
```

Comment:
```suggestion
  - if a permanent error is encountered such that the mutation will never be appliable, the endpoint must ignore that mutation and increment the `lastMutationID` as if it were applied. Failing to do this will deadlock the client if a mutation will always fail.
```